File: web-src/src/app/shared-modules/main/enums/PhoneType.ts
export enum PhoneType {
    HOME_PHONE_1 = 0,
    WORK_PHONE = 1,
    MOBILE_PHONE = 2,
    HOME_PHONE_2 = 3,
    PRIMARY_PHONE = 5,
    NON_PRIMARY_PHONE = 6,
    BUSINESS_PHONE = 10,
    GOOGLE_VOICE = 15,
    OTHER_PHONE = 20,
    REFERENCE = 99
}

export function getPhoneTypeLabel(type: PhoneType): string {
    switch (type) {
        case PhoneType.HOME_PHONE_1:
            return "Home Phone";
        case PhoneType.WORK_PHONE:
            return "Work Phone";
        case PhoneType.MOBILE_PHONE:
            return "Mobile Phone";
        case PhoneType.HOME_PHONE_2:
            return "Other Home Phone";
        case PhoneType.PRIMARY_PHONE:
            return "Primary Phone";
        case PhoneType.BUSINESS_PHONE:
            return "Business Phone";
        case PhoneType.GOOGLE_VOICE:
            return "Google Voice";
        case PhoneType.OTHER_PHONE:
            return "Other Phone";
        case PhoneType.REFERENCE:
            return "Reference";
        default:
            return "Unknown";
    }
}

export const PhoneTypeKeyValueArray = [
    {id: PhoneType.HOME_PHONE_1, label: getPhoneTypeLabel(PhoneType.HOME_PHONE_1)},
    {id: PhoneType.WORK_PHONE, label: getPhoneTypeLabel(PhoneType.WORK_PHONE)},
    {id: PhoneType.MOBILE_PHONE, label: getPhoneTypeLabel(PhoneType.MOBILE_PHONE)},
    {id: PhoneType.HOME_PHONE_2, label: getPhoneTypeLabel(PhoneType.HOME_PHONE_2)},
    {id: PhoneType.PRIMARY_PHONE, label: getPhoneTypeLabel(PhoneType.PRIMARY_PHONE)},
    {id: PhoneType.BUSINESS_PHONE, label: getPhoneTypeLabel(PhoneType.BUSINESS_PHONE)},
    {id: PhoneType.GOOGLE_VOICE, label: getPhoneTypeLabel(PhoneType.GOOGLE_VOICE)},
    {id: PhoneType.OTHER_PHONE, label: getPhoneTypeLabel(PhoneType.OTHER_PHONE)},
    {id: PhoneType.REFERENCE, label: getPhoneTypeLabel(PhoneType.REFERENCE)}
];

File: src/main/java/com/amp/crm/constants/PhoneType.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.amp.crm.constants;

/**
 *
 * @author David
 */
public class PhoneType {

    public static final int PHONE_TYPE_HOME_PHONE_1 = 0;
    public static final int PHONE_TYPE_HOME_PHONE_2 = 3;
    public static final int PHONE_TYPE_MOBILE_PHONE = 1;
    public static final int PHONE_TYPE_WORK_PHONE = 2;
    public static final int PHONE_TYPE_PRIMARY_EMPLOYER_PHONE = 5;
    public static final int PHONE_TYPE_NON_PRIMARY_EMPLOYER_PHONE = 6;
    public static final int PHONE_TYPE_BUSINESS_PHONE = 10;
    public static final int PHONE_TYPE_GOOGLE_VOICE = 15;
    public static final int PHONE_TYPE_OTHER_PHONE = 20;

  

    public static String getPhoneTypeDesc(int phoneType) {
        switch (phoneType) {
            case PHONE_TYPE_HOME_PHONE_1:
                return "Home Phone 1";
            case PHONE_TYPE_HOME_PHONE_2:
                return "Home Phone 2";
            case PHONE_TYPE_MOBILE_PHONE:
                return "Mobile Phone";
            case PHONE_TYPE_WORK_PHONE:
                return "Work Phone";
            case PHONE_TYPE_OTHER_PHONE:
                return "Other Phone";
            case PHONE_TYPE_PRIMARY_EMPLOYER_PHONE:
                return "Primary Employer Phone";
            case PHONE_TYPE_NON_PRIMARY_EMPLOYER_PHONE:
                return "Non-Primary Employer Phone";
            case PHONE_TYPE_GOOGLE_VOICE:
                return "Google Voice";
            case PHONE_TYPE_BUSINESS_PHONE:
                return "Business Phone";
        }
        return "Unknown Phone Type";

    }

}
